### DISCUSSION TITLE
Is the costtime linear with the number of partitions?

### DISCUSSION THREAD

**Discussion by CrazyTianC** [milvus-io/milvus#4596] [Q&A and General discussion] [answered]
I use the default partitions, with 500,000 vectors, it cost 30ms.
I create two partitions, each has 500,000 vectors, it cost 60ms; three partitions cost 90ms.

In practice, I would like to search with tags and full search at the same time. Can it  be calculated  on diffrenrent partitions in parallel?
How can I configure the settings, or how can I write the code to reduce the costtime, if I use multi partitions?

Thanks

**Comment by shiyu22** [comment]
Hi, which version are you using? Milvus supports specifying `partition_tags` when search, if not all partitions in the collection are retrieved by default, you can refer to the [search API](https://milvus.io/docs/v0.10.5/search_vector_python.md).

Regarding performance, multiple partitions of the same amount of data retrieval performance are linear, in fact, partitions and collections are similar, you can refer to [this article](https://www.milvus.io/cn/blogs/2019-12-16-table-partition.md). So we prefer to retrieve a collection with large data, for example, you said that both partitions have 500,000 data, you can try to retrieve a collection of one million, the performance should be faster.

**Reply by CrazyTianC** [reply]
Thanks.
The test is based on 0.11.0. And now I find you rollback to version 0.10.5. I tried 0.10.5. This time the qps is much higher than 0.11.0.

500,000 vectors, with hnsw, 48cores, 64GB ram
version 0.11.0: 30ms for single query; and 30QPS
version 0.10.5: 6ms for single query; and almost 750QPS

what is the reason for this result? I have read that 0.10.5 could combine the queries into one query. I think it is one of the reasons,but is that the main one?

**Reply by shiyu22** [reply]
For QPS, this is really one reason. Also because the data structures of 0.11 and 0.10 are completely different, 0.10 is more stable while 0.11 still has some bugs, so the performance is not as good as 0.10.

**Comment by yhmo** [comment, marked as answer]
Search time cost mostly depends on this factors:
1. how many segments(more segments bring more cost on meta service)
2. how many vectors
3. vector dimensions
4. what kind of index, and index parameters
5. search parameters: nq, nprobe, topk. (For 0.10.x, milvus can combine multiple requests with similar search parameters)
6. CPU ability(sometimes affected by NUMA)
